lldb/include/lldb/lldb-private-types.h

```diff
@@ -121,6 +121,7 @@ typedef struct type256 { uint64_t x[4]; } type256;
 using ValueObjectProviderTy =
     std::function<lldb::ValueObjectSP(ConstString, StackFrame *)>;
 
+typedef int DebuggerDestroyCallbackToken;
```

Comment:
There's no need to make separate lldb_private & SB versions of the CallbackToken.  Since this is intended to be a public type (it shows up in the SB API) you can put it in the lldb namespace - in lldb-types.h, and then both the SB API's and the lldb_private one can use that.